Replace #[doc] error comments with /// type

KnoxSamuel opened the pull request:
Fixed issue #4313, 12 comments changed

## Changes
This commit updates error doc comments with /** */.
It also moves or removes extraneous skip attributes.

    modified:   src/cpu-template-helper/src/template/verify/mod.rs
    modified:   src/vmm/src/cpu_config/x86_64/cpuid/mod.rs        
    modified:   src/vmm/src/cpu_config/x86_64/cpuid/normalize.rs  
    modified:   src/vmm/src/devices/virtio/net/tap.rs
    modified:   src/vmm/src/mmds/mod.rs
    modified:   src/vmm/src/mmds/token.rs
    modified:   src/vmm/src/persist.rs
    modified:   src/vmm/src/vmm_config/machine_config.rs
    modified:   src/vmm/src/vmm_config/mmds.rs

## Reason

Issue #4313 requested changes to error comment types across the codebase.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [X] If a specific issue led to this PR, this PR closes the issue.
- [X] The description of changes is clear and encompassing.
- [X] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

KnoxSamuel commented:
I was trying to fix my author name (I only had my GitHub username KnoxSamuel set on my profile without a actual name set). But when I did an amend and rebase, it added my sign-off to Johnathan's commit. I then rebased again to fix his commit msg, and now the history says 'JonathanWoollett-Light
 authored and
 KnoxSamuel
 committed
' ... is this an issue?

roypat commented on src/cpu-template-helper/src/template/verify/mod.rs:
```diff
@@ -23,7 +23,7 @@ pub enum VerifyError {
     KeyNotFound(String),
     #[rustfmt::skip]
-    #[doc = "Value for {0} mismatched.\n{1}"]
+    /** Value for {0} mismatched.\n{1} */
```
```suggestion
    /** Value for {0} mismatched.\
    {1} */
```

roypat commented on src/vmm/src/persist.rs:
```diff
@@ -158,7 +158,7 @@ pub enum CreateSnapshotError {
     DirtyBitmap(VmmError),
     #[rustfmt::skip]
-    #[doc = "The virtio devices use a features that is incompatible with older versions of Firecracker: {0}"]
+    /** The virtio devices use a features that is incompatible with older versions of Firecracker: {0} */
```
```suggestion
    /** The virtio devices use a features that is incompatible with older\
    versions of Firecracker: {0} */
```

roypat reviewed (commented):
Hi @KnoxSamuel, thanks for tackling this one! 
I think it would be nice if we could wrap the comments, part of the advantage of using the block-doc comments is that we no longer need to have everything on a single line. I've included one example of what I mean as a review comment. Also, could we hoist all the `#[rustfmt::skip]` directives to the top of the enums?

roypat commented:
> I was trying to fix my author name (I only had my GitHub username KnoxSamuel set on my profile without a actual name set). But when I did an amend and rebase, it added my sign-off to Johnathan's commit. I then rebased again to fix his commit msg, and now the history says 'JonathanWoollett-Light authored and KnoxSamuel committed' ... is this an issue?

Mhh, I dont think Jonathan's commit should be a part of this PR at all. Can you drop it and rebase against main again? :o

KnoxSamuel commented:
Thanks for the feedback. @roypat 

I've fixed my fork's history, moved the skip attributes & wrapped those comments to col 75 or less. Does everything look good now? :)

ShadowCurse reviewed (commented):
It seems the `\` at the end is preserved in the comments. Also making comments multi-line makes them to have a `\n` and this breaks some tests.
I think the simplest things we can do are:
- make comments one line again
- use `///` for them (because if the are one line why use `//* */`)
- leave `#[rustfmt::skip]` at the top of structs to prevent long lines being reformatted into multi-lines.

KnoxSamuel commented:
Okay no problem. 
1. Should I rebase my commit or push a new commit to my branch?
2. Undo `#[rustfmt::skip]` changes where they were before, right above the structs/variants?

ShadowCurse commented:
> Should I rebase my commit or push a new commit to my branch?

I think you can rebase. Keep all changes in one commit in this case is fine.

> Undo #[rustfmt::skip] changes where they were before, right above the structs/variants?

You can keep them above the structs, so same way they are currently in the PR.

KnoxSamuel commented:
Let me know if there's anything I missed now!

ShadowCurse commented on src/cpu-template-helper/src/template/verify/mod.rs:
```diff
@@ -22,8 +22,7 @@ pub use x86_64::verify;
     KeyNotFound(String),
-    #[rustfmt::skip]
-    #[doc = "Value for {0} mismatched.\n{1}"]
+    /// Value for {0} mismatched.\n{1}
```
Ok, the last thing. As it appears `#[doc = "Value for {0} mismatched.\n{1}"]` and `/// Value for {0} mismatched.\n{1}` are not equal. The later converts `\n` into `\\n`. So in this case let's use the `/** */`. Never though Rust comments with new lines will be such a pain.
```diff
-    /// Value for {0} mismatched.\n{1}
+    /** Value for {0} mismatched.
+    {1} */
```

KnoxSamuel replied:
Haha strange. Thanks for the help.
